### PR TITLE
fix(cli): profile AssumeRole credentials don't work via proxy

### LINKED ISSUE #7265: Failure to determine Default AWS Account
Post upgrade from 1.28, any cdk synth or deploy commands fail with "Unable to determine the default AWS account", this then fails to create the CDK_DEFAULT_ACCOUNT env variable. We are on a corporate network behind a HTTP proxy with no other egress options. This was functional up to version 1.28, any version since has caused the same issue.

### Reproduction Steps

```
cdk init app --language python
pip install -r requirements.txt
AWS_PROFILE=default cdk synth --trace --verbose
```

### Error Log

```
CDK toolkit version: 1.32.0 (build 9766ad6)
Command line arguments: {
  _: [ 'synth' ],
  verbose: true,
  v: true,
  trace: true,
  'ignore-errors': false,
  ignoreErrors: false,
  json: false,
  j: false,
  ec2creds: undefined,
  i: undefined,
  'version-reporting': undefined,
  versionReporting: undefined,
  'path-metadata': true,
  pathMetadata: true,
  'asset-metadata': true,
  assetMetadata: true,
  'role-arn': undefined,
  r: undefined,
  roleArn: undefined,
  staging: true,
  'no-color': false,
  noColor: false,
  fail: false,
  '$0': '../node_modules/.bin/cdk'
}
Determining whether we're on an EC2 instance.
Does not look like EC2 instance.
Using proxy server: http://corpproxy.domain:1234
cdk.json: {
  "app": "python3 app.py"
}
cdk.context.json: {
  "@aws-cdk/core:enableStackNameDuplicates": "true",
  "aws-cdk:enableDiffNoFail": "true"
}
merged settings: {
  versionReporting: true,
  pathMetadata: true,
  output: 'cdk.out',
  app: 'python3 app.py',
  context: {},
  tags: [],
  assetMetadata: true,
  toolkitBucket: {},
  staging: true
}
Toolkit stack: CDKToolkit
Setting "CDK_DEFAULT_REGION" environment variable to us-east-1
Resolving default credentials
Unable to determine the default AWS account: Error: Client network socket disconnected before secure TLS connection was established
    at connResetException (internal/errors.js:570:14)
    at TLSSocket.onConnectEnd (_tls_wrap.js:1361:19)
    at Object.onceWrapper (events.js:299:28)
    at TLSSocket.emit (events.js:215:7)
    at TLSSocket.EventEmitter.emit (domain.js:476:20)
    at endReadableNT (_stream_readable.js:1183:12)
    at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  message: 'Client network socket disconnected before secure TLS connection was established',
  code: 'NetworkingError',
  path: null,
  host: 'sts.amazonaws.com',
  port: 443,
  localAddress: undefined,
  region: 'us-east-1',
  hostname: 'sts.amazonaws.com',
  retryable: true,
  time: 2020-04-08T22:34:21.259Z
}
context: {
  '@aws-cdk/core:enableStackNameDuplicates': 'true',
  'aws-cdk:enableDiffNoFail': 'true',
  'aws:cdk:enable-path-metadata': true,
  'aws:cdk:enable-asset-metadata': true
}
outdir: cdk.out
env: {
  CDK_DEFAULT_REGION: 'us-east-1',
  CDK_CONTEXT_JSON: '{"@aws-cdk/core:enableStackNameDuplicates":"true","aws-cdk:enableDiffNoFail":"true","aws:cdk:enable-path-metadata":true,"aws:cdk:enable-asset-metadata":true}',
  CDK_OUTDIR: 'cdk.out',
  CDK_CLI_ASM_VERSION: '1.21.0',
  CDK_CLI_VERSION: '1.32.0'
}
Traceback (most recent call last):
  File "app.py", line 8, in <module>
    account=os.environ["CDK_DEFAULT_ACCOUNT"],
  File "/Users/myuser/.asdf/installs/python/3.7.5/lib/python3.7/os.py", line 679, in __getitem__
    raise KeyError(key) from None
KeyError: 'CDK_DEFAULT_ACCOUNT'
Subprocess exited with error 1
Error: Subprocess exited with error 1
    at ChildProcess.<anonymous> (/Users/myuser/code/gitlab/cdk/node_modules/cdk/node_modules/aws-cdk/lib/api/cxapp/exec.ts:115:23)
    at ChildProcess.emit (events.js:210:5)
    at ChildProcess.EventEmitter.emit (domain.js:476:20)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)

```

### Environment

  - **CLI Version      :** 1.32
  - **Framework Version:** 1.32
  - **OS               :** macOS 10.14.6
  - **Language         :** Python

### Other

We leverage the CDK_DEFAULT_REGION and CDK_DEFAULT_ACCOUNT env variables in the env parameter on each stack call.
```
env = core.Environment(
    account=os.environ["CDK_DEFAULT_ACCOUNT"],
    region=os.environ["CDK_DEFAULT_REGION"]
    )

app = core.App()
TestAppStack(app, "test-app", env=env)

app.synth()
```

Our corporate HTTP proxy spoofs certificates.

Only relative change based on this issue was here:
https://github.com/aws/aws-cdk/commit/a75f711aea3dac83c6feec885b7df9b14a39486e#diff-c59a64c4aaf11c23a64d46cfdd682054L138

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by rix0rrr**
### Commit Message

The proxy configuration was not being properly passed to the
`AWS.SharedIniFileCredentials` object, so if users configure AssumeRole
credentials in their `~/.aws/config` file, it would try to connect to
STS directly instead of going through the proxy.

Properly parse and pass the HTTP options to the right AWS file.

(Note that this object only takes `HTTPOptions` and not
`ConfigurationOptions`, so there's no way to override the user agent
on this initial STS call).

Fixes #7265.

### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk/lib/api/aws-auth/sdk-provider.ts [hunk 1]
```diff
@@ -298,6 +301,7 @@ function defaultHttpOptions(options: SdkHttpOptions) {
     debug('Using proxy server: %s', proxyAddress);
     // eslint-disable-next-line @typescript-eslint/no-require-imports
     const ProxyAgent: any = require('proxy-agent');
+    // tslint:disable-next-line:no-console
```
Comment (shivlaks) [inline packages/aws-cdk/lib/api/aws-auth/sdk-provider.ts:304]: remove?

File: packages/aws-cdk/test/api/sdk-provider.test.ts [hunk 2]
```diff
@@ -138,6 +149,40 @@ describe('CLI compatible credentials loading', () => {
...
+
+  test('even when using a profile to assume another profile, STS calls goes through the proxy', async () => {
+    // Messy mocking
+    let called = false;
+    jest.mock('proxy-agent', () => {
+      class FakeAgent extends require('https').Agent {
+        public addRequest(_: any, __: any) {
+          // This error takes 6 seconds to be completely handled. It might be retries in the SDK
+          // somewhere, or something about the Node event loop. I've spent an hour trying to figure
+          // it out and I can't, and I gave up. We'll just have to live with this.
```
Comment (shivlaks) [inline packages/aws-cdk/test/api/sdk-provider.test.ts:161]: maybe add a `FIXME` here? don't see it as being blocking for now but interesting nonetheless

File: packages/aws-cdk/test/api/sdk-provider.test.ts [hunk 3]
```diff
@@ -138,6 +149,40 @@ describe('CLI compatible credentials loading', () => {
 
     await expect(provider.forEnvironment(`${uid}some_account_#`, 'def', Mode.ForReading)).rejects.toThrow('Need to perform AWS calls');
   });
+
+  test('even when using a profile to assume another profile, STS calls goes through the proxy', async () => {
+    // Messy mocking
+    let called = false;
+    jest.mock('proxy-agent', () => {
+      class FakeAgent extends require('https').Agent {
```
Comment (shivlaks) [inline packages/aws-cdk/test/api/sdk-provider.test.ts:157]: does this not complain about using require?

Comment (rix0rrr) [reply packages/aws-cdk/test/api/sdk-provider.test.ts:158]: Of course it does :)

**Pull request merged**